core/base/inc/Rtypes.h

```diff
@@ -225,44 +225,120 @@ namespace ROOT {
 
 } // namespace ROOT
 
-// The macros below use TGenericClassInfo, so let's ensure it is included
+// The macros below use TGenericClassInfo and TInstrumentedIsAProxy, so let's
+// ensure they are included.
 #ifndef ROOT_TGenericClassInfo
 #include "TGenericClassInfo.h"
 #endif
+#ifndef ROOT_TIsAProxy
+#include "TIsAProxy.h"
+#endif
+
+
+// Parts or ROOT are compiled with no-rtti and cannot use or even see typeid().
+// Hide the following declaration (that is only needed by ClassDefInline) for
+// these cases by #defining R__NO_INLINE_CLASSDEF:
+#ifndef R__NO_INLINE_CLASSDEF
+namespace ROOT {
+   template <typename T>
+   class ClassDefGenerateInitInstanceLocalInjector {
+      static TClass* fgIsA;
+   public:
+      static void *New(void *p) { return p ? new(p) T : new T; };
```

Comment:
Those are not universally applicable.   Depending on whether the default constructor is accessible some or all the routines are inaccessible.   When the user provide an I/O constructor (__void__ or TROOTIOCtor or user defined parameter), it should take precedence.